### DISCUSSION TITLE
Access to RestServer in RestResources class

### DISCUSSION THREAD

**Discussion by dariofil-star** [scottoffen/grapevine#90] [Q&A] [answered]
RestResources in older versions seem to inherit the RestServer (Server property) from a base class.
In the latest build, this feature seems to have disappeared, and RestResource is an attribute.

I run multiple instances of RestServer and need the RestResources to behave a little different for each RestServer instance.

Any hints on how to pass RestServer into RestResources?
Seems a little too complex for me to figure out.

**Comment by dariofil-star** [comment]
Seems Scott has already answered my question on another thread;
# #Helps to read discussion https://github.com/scottoffen/grapevine/discussions/50

You can use a constructor and a field... thats it.

"_Because the instance of IRestServer is already registered in the service collection used by the router (this happens automatically) as a singleton, you don't need to do any more wiring up for this to work._"

**Comment by scottoffen** [comment, marked as answer]
You can inject the server via your constructor, as seen in the [`HelloResource`](https://github.com/scottoffen/grapevine/blob/409fa6304e84d0a7f722b0e45aceb9cfeb1b197b/src/Samples/Resources/HelloResource.cs#L16) in the sample project.

```csharp
private readonly IRestServer _server;

public YourResource(IRestServer server)
{
    _server = server;
}
```
